modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/MovedEventSyncCommand.java

```diff
@@ -31,35 +31,43 @@ protected void doSync()
                     ? contentService.getById( sourceParent.getId() ).getPath()
-                    : sourceRoot.getId().equals( sourceParent.getId() ) ? params.getSourceContent().getPath().getRoot() : null;
+                    : sourceRoot.getId().equals( sourceParent.getId() ) ? ContentPath.ROOT : null;
 
                 if ( targetParentPath != null )
                 {
-                    if ( !targetParentPath.equals( params.getTargetContent().getParentPath() ) )
+                    if ( targetParentPath.equals( params.getTargetContent().getParentPath() ) )
                     {
-                        final ContentPath newPath = buildNewPath( targetParentPath, params.getTargetContent().getName() );
-
-                        if ( !Objects.equals( newPath.getName(), params.getTargetContent().getPath().getName() ) )
+                        final ContentPath newContentPath =
+                            ContentPath.from( targetParentPath, params.getTargetContent().getName().toString() );
+                        final Content existedContent =
+                            contentService.contentExists( newContentPath ) ? contentService.getByPath( newContentPath ) : null;
+                        if ( existedContent != null && existedContent.getId().equals( params.getTargetContent().getId() ) )
                         {
-                            contentService.rename( RenameContentParams.create()
-                                                       .contentId( params.getTargetContent().getId() )
-                                                       .newName( ContentName.from( newPath.getName() ) )
-                                                       .build() );
+                            return; // content is already there.
                         }
+                    }
 
-                        contentService.move( MoveContentParams.create()
-                                                 .contentId( params.getTargetContent().getId() )
-                                                 .parentContentPath( targetParentPath )
-                                                 .stopInherit( false )
-                                                 .build() );
+                    final ContentPath newPath = buildNewPath( targetParentPath, params.getTargetContent().getName() );
 
+                    if ( !Objects.equals( newPath.getName(), params.getTargetContent().getPath().getName() ) )
+                    {
+                        contentService.rename( RenameContentParams.create()
+                                                   .contentId( params.getTargetContent().getId() )
+                                                   .newName( ContentName.from( newPath.getName() ) )
+                                                   .build() );
                     }
+
+                    contentService.move( MoveContentParams.create()
```

Comment:
Create new issue to do renmov in one command

Reply:
https://github.com/enonic/xp/issues/9074